### PR TITLE
Bwt 349: park-access-statuses API endpoint filter for accessStatus field

### PR DESCRIPTION
**Pull request by nataliasabadysh**
Jira Ticket:
Bwt 349

Jira Ticket URL:
https://bcparksdigital.atlassian.net/jira/software/projects/BWT/boards/5?assignee=633db6bb61dbef2805c1f25c&selectedIssue=BWT-349

### CODE REVIEW

**Comment by molund** [comment]
@nataliasabadysh would it be fairly easy to also add support for a `accessStatus_ne` parameter?
e.g. `?accessStatus_ne=Open`
I think this option might accomplish the original goal of the ticket description better than checking for 9 different values.

1. Open
2. Full closure
3. Closed to the public
4. Inaccessible
5. Park is under evacuation alert
6. Park is inaccessible
7. Restricted; permit required
8. Park is under evacuation order
9. Park is within a fire perimeter
10. Warning

**Review by molund — CHANGES_REQUESTED** [review]
@nataliasabadysh would it be fairly easy to also add support for a `accessStatus_ne` (not equal to) parameter?
e.g. `?accessStatus_ne=Open`
I think this option might accomplish the original goal of the ticket description better than checking for 9 different values.

File: src/cms/api/protected-area/custom/protected-area-status.js [hunk 1]
```diff
@@ -270,6 +272,17 @@ const getProtectedAreaStatus = async (ctx) => {
...
+      (o) => o?.accessStatus?.toLowerCase() == accessStatus.toLowerCase()
+    );
+  } else if (accessStatus_ne) {
+    return payload = payload.filter(
+      (o) => o?.accessStatus?.toLowerCase() != accessStatus_ne.toLowerCase()
+    );
+  }
+
   // Store unfiltered payload into a cached location (unfiltered == no orcs specified)
   if (!ctx.query.orcs) {
```
Comment (molund) [inline src/cms/api/protected-area/custom/protected-area-status.js:287]: I think we need to update this to include accessStatus and accessStatus_ne (in additon to orcs).

**Review by molund — CHANGES_REQUESTED** [review]

File: src/cms/api/protected-area/custom/protected-area-status.js (hunk 1 above)
Comment (nataliasabadysh) [reply src/cms/api/protected-area/custom/protected-area-status.js:298]: yes, I added a case with the both 
please have a look 
thank you

**Review by ayumi-oxd — APPROVED** [review]
LGTM, if you modify the condition part in the filter that @molund mentioned in the meeting.

**Review by molund — APPROVED** [review]

**Review by molund — CHANGES_REQUESTED** [review]
This if condition should be updated so the ParkAccessStatusCaches collection only gets updated when there is an unfiltered query.....
` if (!ctx.query.orcs) {`

There is a cron job that runs every 5 minutes to update ParkAccessStatusCaches

**Review by molund — APPROVED** [review]

**Pull request merged**